config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+##
+# See https://github.com/kickstarter/rack-attack/blob/master/docs/example_configuration.md
+# for more configuration options
+
+# If RACK_ATTACK_REDIS_URL is set in the environment, turn on throttling and use Redis as the cache store
+if ENV['RACK_ATTACK_REDIS_URL']
+  Rack::Attack.cache.store = ActiveSupport::Cache::RedisCacheStore.new(url: ENV['RACK_ATTACK_REDIS_URL'])
+  ### Throttle Spammy Clients ###
+
+  # If any single client IP is making tons of requests, then they're
+  # probably malicious or a poorly-configured scraper. Either way, they
+  # don't deserve to hog all of the app server's CPU. Cut them off!
+  #
+  # Note: If you're serving assets through rack, those requests may be
+  # counted by rack-attack and this throttle may be activated too
+  # quickly. If so, enable the condition to exclude them from tracking.
+
+  # Throttle all requests by IP (60rpm)
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:req/ip:#{req.ip}"
+  Rack::Attack.throttle('req/ip', limit: 300, period: 5.minutes, &:ip)
+
+  # Throttle search requests by IP (15rpm)
+  Rack::Attack.throttle('req/search/ip', limit: 15, period: 1.minute) do |req|
+    route = begin
+      Rails.application.routes.recognize_path(req.path) || {}
+    rescue StandardError
+      {}
+    end
+
+    req.ip if route[:controller] == 'catalog' && %w[index facet].include?(route[:action])
+  end
+
+  # Throttle document actions requests by IP (15rpm)
+  Rack::Attack.throttle('req/actions/ip', limit: 15, period: 1.minute) do |req|
```

Comment:
Maybe purposeful, but this block's outer logic is the same as above, could maybe be combined.